src/Controllers/DatabaseController.js

```diff
@@ -192,15 +192,79 @@ const validateQuery = (
 
 // Filters out any data that shouldn't be on this REST-formatted object.
 const filterSensitiveData = (
-  isMaster,
-  aclGroup,
-  className,
-  protectedFields,
-  object
+  isMaster: boolean,
+  aclGroup: any[],
+  auth: any,
+  operation: any,
+  schema: SchemaController.SchemaController,
+  className: string,
+  protectedFields: null | Array<any>,
+  object: any
 ) => {
-  protectedFields && protectedFields.forEach(k => delete object[k]);
+  let userId = null;
+  if (auth && auth.user) userId = auth.user.id;
 
-  if (className !== '_User') {
+  // replace protectedFields when using pointer-permissions
+  const perms = schema.getClassLevelPermissions(className);
+  if (perms) {
+    const field =
+      ['get', 'find'].indexOf(operation) > -1
+        ? 'readUserFields'
+        : 'writeUserFields';
+    const fieldKeys: string[] = perms[field];
+
+    if (
+      field === 'readUserFields' &&
```

Comment:
Does it only work together with CLP? I mean, if the CLP is set to public, shouldn't the owner be able to see their protected fields?

Reply:
I'm not exactly sure if I understand your question. But as far as I understand the `protectedFields` were introduced to allow overriding permissions like public read. When not specifying `protectedFields` the owner (defined by pointer-permissions) of an object is able to read all fields on the object. If an user chooses to hide any property using the protectedFields for spefici a role or an user/user-array specified by pointer permissions (introduced by this pr), then the field should be hidden.

For the `_User` class the setup is different. There is code in place to allow the user reading all properties of the corresponding object in the `_User` collection/table. This was already in place before this pr.

Reply:
That's actually not my question. Checking your code and your examples, it seems that I always need to define the `readUserFields` in the CLP to have the `readUserFields:someUserField` working in the protected fields. But I think that those two things should work independently. For example, the test below should pass:

```js
  it('should allow access using single user pointer-permissions', async done => {
        const config = Config.get(Parse.applicationId);
        const obj = new Parse.Object('AnObject');

        obj.set('owner', user1);
        obj.set('test', 'test');
        await obj.save();

        const schema = await config.database.loadSchema();
        await schema.updateClass(
          'AnObject',
          {},
          {
            get: { '*': true },
            find: { '*': true },
            // readUserFields: ['owner'], ---> It shouldn't be required since the public read is already specified
            protectedFields: { '*': ['owner'], 'readUserFields:owner': [] },
          }
        );

        await Parse.User.logIn('user1', 'password');
        const objectAgain = await obj.fetch();
        expect(objectAgain.get('owner').id).toBe(user1.id);
        expect(objectAgain.get('test')).toBe('test');
        done();
      });
```

Reply:
You are correct by assuming that `readUserFields` currently has to be set in the CLP. I do like your approach of separating the `readUserFields` CLP and protectedFields.
But the name `readUserFields` in CLP suggests that users stored in the field are able to **read** all (user-) fields. I do think that we need a different prefix for the protectedFields as this is not directly related to the `readUserFields` anymore.
After separating the two options the only similarity is the way users a stored in the referenced column (Pointer<_User> or Array<Pointer<_User>>).

Reply:
`userField:someField` looks good to me!

Reply:
Decoupled `readUserField` CLP and renamed `userField` as discussed and updated tests to reflect changes.